tests/Doctrine/Tests/DBAL/Functional/Schema/MySqlSchemaManagerTest.php

```diff
@@ -319,4 +319,89 @@ public function testListFloatTypeColumns()
         $this->assertFalse($columns['col']->getUnsigned());
         $this->assertTrue($columns['col_unsigned']->getUnsigned());
     }
+
+    /**
+     * Test default null values (especially for mariadb 10.2.7+)
+     * @link https://github.com/doctrine/dbal/issues/2817
+     */
+    public function testNullColumnDefaultWithTable()
+    {
+
+        $table = new Table("test_column_defaults_with_table");
+        $table->addColumn('col0', 'integer', ['notnull' => false]);
+        $table->addColumn('col1', 'integer', ['notnull' => false, 'default' => null]);
+        $table->addColumn('col2', 'string', ['notnull' => false, 'default' => null]);
+        $table->addColumn('col3', 'string', ['notnull' => false, 'default' => 'NULL']);
+        $table->addColumn('col4', 'string', ['notnull' => false, 'default' => 'Hello world']);
+        $table->addColumn('col5', 'datetime', ['notnull' => false, 'default' => null]);
+        $table->addColumn('col6', 'decimal', ['scale' => 3, 'precision' => 6, 'notnull' => false, 'default' => -2.3]);
+        $table->addColumn('col7', 'date', ['notnull' => false, 'default' => '2012-12-12']);
+        $table->addColumn('col8', 'string', ['notnull' => true, 'default' => '']);
+        $table->addColumn('col9', 'integer', ['notnull' => false, 'default' => 0]);
+        $table->addColumn('col10', 'string', ['notnull' => false, 'default' => 'He"ll"o world']);
+
+        $this->_sm->dropAndCreateTable($table);
+
+        $onlineTable = $this->_sm->listTableDetails("test_column_defaults_with_table");
+        $this->assertNull($onlineTable->getColumn('col0')->getDefault());
+        $this->assertNull($onlineTable->getColumn('col1')->getDefault());
+        $this->assertNull($onlineTable->getColumn('col2')->getDefault());
+        $this->assertEquals('NULL', $onlineTable->getColumn('col3')->getDefault());
+        $this->assertEquals('Hello world', $onlineTable->getColumn('col4')->getDefault());
+        $this->assertNull($onlineTable->getColumn('col5')->getDefault());
+        $this->assertEquals(-2.3, $onlineTable->getColumn('col6')->getDefault());
+        $this->assertEquals('2012-12-12', $onlineTable->getColumn('col7')->getDefault());
+        $this->assertTrue($onlineTable->getColumn('col8')->getNotnull());
+        $this->assertEquals('', $onlineTable->getColumn('col8')->getDefault());
+        $this->assertEquals(0, $onlineTable->getColumn('col9')->getDefault());
+        $this->assertEquals('He"ll"o world', $onlineTable->getColumn('col10')->getDefault());
+
+        $comparator = new Comparator();
+
+        $diff = $comparator->diffTable($table, $onlineTable);
+        $this->assertFalse($diff, "Tables should be identical with column defaults.");
+    }
+
+    /**
+     * For testing MariaDB 10.2+ / MySQL differences
```

Comment:
Put this description in the method name instead

Reply:
Changed to `testColumnDefaultsWithCreateAreValidForMySqlAndMariaDb102`

Reply:
Afterwards thinking, this test is newly introduced when working on MariaDB. Actually it's valid for both anyway. I'll cleanup the description completely. But good to know about method naming